### ISSUE TITLE
Icons for GUI apps installed in containers should work in the host OS

### ISSUE THREAD

**Issue by markstos** [containers/toolbox#354]
Chrome OS / Crostini has demonstrated how icons for apps installed in containers can be integrated with the host OS: Icons immediately appear in the parent desktop after installation. Clicking the icons causes the app to launch in it's child container if it's not already running. 

A [number of services](https://chromium.googlesource.com/chromiumos/docs/+/master/containers_and_vms.md) running instead and outside the container are used to accomplish this, including Maitred, Cicerone and Garcon.

To make the scope more manageable for now, Google only officially supports this integration for one running container and one operating system (Currently Debian). 

I believe the basic design is that packages inside the contain "watch" for changes in `.desktop` files. When found, the `.desktop` files and related icons are communicated to the host, who installs their company that adjusts the command to reference the container. For reasons that aren't clear, currently Chrome OS only supports `.png` icons for such apps, not `.svg`.

Thankfully Chromium OS is open source so their approach can be reviewed in detail.

**Comment by debarshiray** [comment]
This sounds a lot like replicating things that Flatpak already does and is designed to do.

I know that the presence of `toolbox run ...` makes it tempting to have deeper integration for graphical applications, but we don't want to get into handling desktop files. `toolbox run ...` is meant to facilitate things like creating your own custom desktop file that fires up a graphical Emacs instance running inside a specific container. It's a thin wrapper around `toolbox enter`, but I'd be careful about heading too far in that direction.

In general, Podman (or an OCI container) is a bad option for running GUI apps. That's the whole reason why Flatpak exists. Is there a specific reason or a specific graphical application that you want or need to run inside a Podman container?

I have one pet complaint called *devhelp*. I want the devhelp Flatpak to pick up documentation that might be installed in a Toolbox container's `/usr/share/gtk-doc`, but there might be ways to fix it.

**Comment by markstos** [comment]
I use Chromium's ability to treat a website as a desktop "app" extensively. There was a project to package Chromium as a flatpak but it appears to have stalled: https://github.com/mariospr/chromium-browser-flatpak Maybe I'm missing another flatpak source for this popular app?  I also tried the "snap" for Chromium. Canonical's supporting the snap, but the "web app" feature of Chromium is broken in the snap package. Firefox hasn't included this feature for several years. So I'm interested in running Chromium within a container, with desktop integration. Maybe there's another way to treat a website as a desktop app that I'm missing.

**Comment by debarshiray** [comment]
Chrome is tricky is because it's own sandboxing doesn't work well inside a Flatpak container. Specifically, it tries to create a user namespace and fails.

Here's another attempt to Flatpak Chrome: https://github.com/flathub/flathub/pull/566

Here's a discussion on the topic: https://discourse.gnome.org/t/sandboxing-portal/1651

There's ongoing effort to get Mozilla to publish Flatpaks of Firefox, but until that happens you can use the Flatpaks published by the Fedora maintainers of Firefox from https://firefox-flatpak.mojefedora.cz/

You specifically mentioned web "app" support, which reminds me of Epiphany. Ironically, Epiphany or GNOME Web isn't on Flathub because it's upstream maintainer wasn't happy with the security maintenance of the GNOME runtimes. Fortunately, the Flatpak runtimes published by Fedora do a lot better in this regard and you can get Fedora's Flatpak build of Epiphany. See: https://blog.fishsoup.net/2018/12/04/flatpaks-in-fedora-now-live/

In short, it's an evolving situation. We'll finally get to a situation where the major web browsers are widely available as Flatpaks with a tight sandbox, but we aren't there yet.

**Comment by markstos** [comment]
@debarshiray The Gnome Web Flatpak seems like a good fit for me. Thanks for the recommendation!

**Comment by debarshiray** [comment]
I am glad that you found it useful. :)

Closing.

**Comment by debarshiray** [comment]
Duplicate of https://github.com/containers/toolbox/issues/417
